lxd/instance/drivers/driver_lxc.go

```diff
@@ -3886,7 +3886,7 @@ func (d *lxc) Update(args db.InstanceArgs, userRequested bool) error {
 
 	if userRequested {
 		// Validate the new config
-		err := instance.ValidConfig(d.state.OS, args.Config, false, false)
+		err := instance.ValidConfig(d.state.OS, args.Config, false, d.dbType)
```

Comment:
@stgraber not related to this PR really, but I've wondered in the past and now, why do we run validation of instance config and devices twice during `Update()`, once with expanded = false and just instance config, and once with expanded = true with profile config applied. I know what the expanded checks are more expensive to do, but we still do them further down, so I don't see the point of doing the non-expanded ones here too. Thanks

Reply:
Oh actually I know why for this change, its because we want to ensure that only instance type specific config is allowed in the instance's actual config, but the expanded config with the profile config could contain keys for any instance type.

Please could you add comments to that effect for future readers as its quite a subtle thing. Thanks

Reply:
It did feel a bit all over the place, though these days we certainly do need to test both individually:
 - Instance local config is validated based on its type and so will not allow VM keys on container or vice-versa
 - Instance expanded config is validated without checking the instance type and allows for checking that conflicting keys or cross-key dependencies are satisfied.

I was tempted to change the validation function to take in an Instance and just do everything in one shot but this wasn't practical to do because of us using the same functions from profiles.

Anyway, something we can improve later on for sure.